### ISSUE TITLE
Assign source IP Address

### ISSUE THREAD

**Issue by gt5700** [MichaCo/DnsClient.NET#35]
I have a multihomed (multi-interface) server, and I need to ensure DNS requests are sourced from a specific IP Address on the server. The default for Udp/TcpClient is to source from the address "closest" from a routing point of view to the destination address. I need it to be sourced from a fixed IP Address regardless of which interface the request "leaves" the server on.

If this helps, for other code, I've used the UdpClient/TcpClient constructors that take an IPEndpoint allowing you to specify the local address the client binds to.

eg/
var udpClient = new UdpClient(new IPEndPoint(srcIPAddress,0));

**Comment by MichaCo** [comment]
Yeah we could add that as an additional configuration. Sounds reasonable.
